### PR TITLE
#151515982 enhance UI and implement feedback

### PR DESCRIPTION
**Pull request by AnthonyGW**
#### What does this PR do?
Improve on the UI made in development
Implement feedback received in previous PullRequests, Code Review and Code Defense.

#### Description of Task to be completed?
Use bootstrap to style the pages and forms.
Feedback Implemented:
1. Add comments to code.
2. Break long lines
3. Add newline to end of files
4. Remove unused imports

#### How should this be manually tested?
There are no test files implemented yet.
Test functionality by interacting with the GUI forms and buttons, the described routes have respective links to go to them:
1. On the base route (`/`), Click on the button `Register` to use create a new user by providing a username, email address and password in the form provided. App should open the Login form when successful.
2. Log in with account email address and password. App should redirect to user dashboard when successful.
3. Create a new bucketlist with the form to the left of the page.
4. View the bucketlists created to the right of the page.
5. Search for bucketlists by name using the search bar.
6. View the rest of the user's bucketlists (when they are more than 10) using the page navigation buttons.
7. Edit or delete a specific bucketlist using the links in the same row. User can leave the form fields blank to keep the same information when editing.
8. View a specific bucketlist's items by clicking on its name.
9. Create a new bucketlist item using the form to the left of the page.
10. View bucketlist items to the right of the page.

#### Any background context you want to provide?
The UI is based on the challenge 2 bootcamp project which can be found here: [BucketListUltima](https://github.com/AnthonyWaithaka/Bucket-List-Ultima)

Pivotal Tracker story: <https://www.pivotaltracker.com/story/show/151513228>

#### Screenshots
screen shot 2017-10-03 at 16 26 58

screen shot 2017-10-03 at 16 27 07

screen shot 2017-10-03 at 16 27 17

screen shot 2017-10-03 at 16 27 36

screen shot 2017-10-03 at 16 28 22

### CODE REVIEW

**Comment by AnthonyGW** [comment]
Remaining with validating form inputs to avoid double entries/ invalid entries and a toggle for the search bar in bucketlists view to search for either bucketlists or bucketlist items.

File: src/Components/nav.jsx [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-                            <li class={loginClass}>
-                                <Link to="/login" onClick={this.toggleCollapse.bind(this)}>Log In</Link>
-                            </li>
-                        </ul>
-                    </div>
-                </div>
-            </nav>
-        )
-    }
-}
```
Comment (jonathankamau) [inline src/Components/nav.jsx]: new line needed here

File: src/actions/itemactions.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// src/actions/itemactions.js
+// Actions defined here as functions can be called throughout the app to operate on the bucketlist
```
Comment (jonathankamau) [inline src/actions/itemactions.js:2]: I see that the operations defined in this file have to do with the bucketlist items not bucketlist.

File: src/pages/bucketlist_items.jsx [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+            editid: null,
+            name: "",
+            description: ""
+        });
+        document.getElementById("create-item-form").reset();        
+    }
+
+    render(){
+        let bucketlistItems = this.state.bucketlist_items;
+        console.log("captured items: ", bucketlistItems);
```
Comment (jonathankamau) [inline src/pages/bucketlist_items.jsx]: why this `console.log`?

File: src/actions/itemactions.js (hunk 2 above)
Comment (AnthonyGW) [reply src/actions/itemactions.js:2]: The comment continues on the next line to mention the item store

File: src/pages/bucketlist_items.jsx (hunk 3 above)
Comment (AnthonyGW) [reply src/pages/bucketlist_items.jsx]: It was to track the items that were saved in the store and passed to the component. Mostly for development purposes, it's not needed in the production build

**Pull request merged**